### ISSUE TITLE
Clarification about classification heads

### ISSUE THREAD

**Issue by danielm1405** [mlfoundations/task_vectors#10]
Hi @gabrielilharco,
thanks for the great work!

I have a question about classification heads used in your experiments and available [here](https://drive.google.com/drive/folders/1u_Tva6x0p6oxu5Eo0ZZsf-520Cc_3MKw). How exactly did you train them? Looking at [the code](https://github.com/mlfoundations/task_vectors/blob/main/src/heads.py#L13-L51) I can see that that you manually construct a zero shot classifier based on embeddings of class names put in various templates. Importantly, you use the **pretrained** OpenCLIP model to calculate embeddings for classifier, **not** the model **finetuned** for a particular task. Am I right about that?

What is the reason for that? To me, the most natural way to obtain the classifier would be to get `model.classification_head` after the finetuning of a task specific model ([here](https://github.com/mlfoundations/task_vectors/blob/main/src/finetune.py#L109)). This classification head is aligned with the finetuned model while the zeroshot head is aligned with pretrained model therefore the head from finetuning seems more suitable. Did you consider such an approach?

**Comment by gabrielilharco** [comment]
Hi @danielm1405. You are right, the classification heads are built from the pretrained (zero-shot) model. Note that when fine-tuning, _we do not modify the classification head_, so both the pretrained and fine-tuned models share the same classification heads for the a given task. This is done in order to preserve the open-vocabulary nature of the model, so it can still be used for any downstream task. Freezing the classification head also doesn't harm accuracy, as we show in our related work https://arxiv.org/abs/2208.05592 (Appendix C).

**Comment by danielm1405** [comment]
Ok, it is clear now. Thank you for the detailed explanation and providing an additional source.
